include/ur_api.h

```diff
@@ -6619,7 +6619,6 @@ urEnqueueMemUnmap(
 ///         + `patternSize > size`
 ///         + `(patternSize & (patternSize - 1)) != 0`
 ///         + `size % patternSize != 0`
-///         + If `size` is higher than the allocation size of `ptr`
```

Comment:
this is not supported by the adapters, but it could be added to the validation layer as common functionality for all. It would be a matter of querying for the allocations' size and compare against the size passed.